tests/csm/rest/test_iam_users.py

```diff
@@ -1430,3 +1430,187 @@ def test_37783(self):
+        uid1 = uid+"1"
+        payload = {}
+        payload.update({"display_name":uid})
+        response = self.csm_obj.modify_iam_user_rgw(uid1, payload)
+        assert response.status_code == HTTPStatus.NOT_FOUND, "Status code check failed."
+        assert response.json()["error_code"] == resp_error_code, (
+            "Error code check failed.")
+        if CSM_REST_CFG["msg_check"] == "enable":
+            assert response.json()["message"] == msg , "Message check failed."
+        assert response.json()["message_id"] == resp_msg_id, "Message ID check failed."
+        self.log.info("STEP 4: Perform get iam users to verify new display name")
+        get_resp = self.csm_obj.get_iam_user(uid)
+        self.log.info("Print user info %s", get_resp.json())
+        assert_utils.assert_true(get_resp.status_code == HTTPStatus.OK, "Get IAM user failed")
+        self.log.info("STEP 5: Check if update is done successfully")
+        assert_utils.assert_true((get_resp.json()["display_name"] == resp.json()["display_name"]),
+                                 "Display Name updated")
+        self.log.info("[END]Update User with display name, non existing uid in request")
+        self.log.info("##### Test completed -  %s #####", test_case_name)
+
+    @pytest.mark.csmrest
+    @pytest.mark.lc
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.parallel
+    @pytest.mark.tags('TEST-38089')
+    def test_38089(self):
+        """
+        Update request with uid and generate-key
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("[START] STEP 1: Creating IAM user with basic parameters")
+        payload = self.csm_obj.iam_user_payload_rgw("random")
+        self.log.info("payload :  %s", payload)
+        resp1 = self.csm_obj.create_iam_user_rgw(payload)
+        assert_utils.assert_true(resp1.status_code == HTTPStatus.CREATED, "IAM user creation failed")
+        uid = resp1.json()["tenant"] + "$" + payload['uid']
+        self.created_iam_users.add(uid)
+        self.log.info("STEP 2: Perform get iam users")
+        get_resp = self.csm_obj.get_iam_user(uid)
+        assert_utils.assert_true(get_resp.status_code == HTTPStatus.OK, "Get IAM user failed")
+        self.log.info("STEP 3: Send request to update uid and generate-key")
+        payload = {}
+        payload.update({"generate_key":True})
+        resp = self.csm_obj.modify_iam_user_rgw(uid, payload)
+        self.log.info("STEP 4: Perform get iam users to verify new display name")
+        get_resp = self.csm_obj.get_iam_user(uid)
+        self.log.info("Print user info %s", get_resp.json())
+        assert_utils.assert_true(get_resp.status_code == HTTPStatus.OK, "Get IAM user failed")
+        self.log.info("STEP 5: Check if keys are generated successfully")
+        assert_utils.assert_true(get_resp.json()["keys"][0]["access_key"]!="",
```

Comment:
here we should expect new keys generated other than existing one before patch request

Reply:
Check for second keys pair generation added and comparison is being done with new keys.